src/nix/diff-closures.md

```diff
@@ -24,28 +24,50 @@ of packages, as well as changes in store path sizes.
 
 For each package name in the two closures (where a package name is
 defined as the name component of a store path excluding the version),
-if there is a change in the set of versions of the package, or a
-change in the size of the store paths of more than 8 KiB, it prints a
-line like this:
+it returns information about the change in the set of versions of the
+package, and the change in the size of the store paths.
 
-```console
-dolphin: 20.08.1 → 20.08.2, +13.9 KiB
-```
-
-No size change is shown if it's below the threshold. If the package
-does not exist in either the *before* or *after* closures, it is
-represented using `∅` (empty set) on the appropriate side of the
-arrow. If a package has an empty version string, the version is
-rendered as `ε` (epsilon).
+If the package does not exist in either the *before* or *after*
+closures, it is represented using `∅` (empty set) on the appropriate
+side of the arrow. If a package has an empty version string, the
+version is rendered as `ε` (epsilon).
 
 There may be multiple versions of a package in each closure. In that
 case, only the changed versions are shown. Thus,
 
 ```console
-libfoo: 1.2, 1.3 → 1.4
+1.2, 1.3 → 1.4
 ```
 
 leaves open the possibility that there are other versions (e.g. `1.1`)
 that exist in both closures.
 
+## Regular output
+
+The regular output prints lines like this:
+
+```console
+dolphin: 20.08.1 → 20.08.2, +13.9 KiB
+```
+
+No size change is shown if it's below the 8 KiB threshold.
+
+## JSON output
+
+With `--json`, the output is in a JSON representation suitable for automatic
+processing by other tools.
+
+The printed result looks like this:
+
+```json
+{
+  "dolphin": {
+    "after": "20.08.1",
+    "before": "20.08.2",
+    "sizeDelta": 13900
+  },
```

Comment:
I share the concern that @SuperSandro2000 originally raised about these symbols not being typable on most keyboards, but I also understand that just `null` is not explicit enough here because there are two different ways of a version not existing. As `"after"` and `"before"` can also contain multiple versions separated by commas, I would propose the following solution:

Make `"after"` and `"before"` lists of strings. `∅` (aka no version of this package exists/existed) is represented by an empty array `[]`, and `ε` (aka an empty version of this package exists/existed) is represented by an array containing a single empty string `[ "" ]`.

I feel this would make the output much more idiomatic to the semantics of JSON.

If you decide not to do this, I think it would at least be useful to add examples with `∅`, `ε` and multiple versions here to show that the JSON output is exactly like the printed one.

Reply:
I like the idea.

However, I feel we should keep the two outputs (JSON, and regular print) consistent.

What do the other maintainers think about changing `ε` for `[ "" ]` and `∅` `for `[]` in the old display?